### ISSUE TITLE
Add firebase-storage

### ISSUE THREAD

**Issue by fuljo** [firebase/polymerfire#10]
### Description

Please add a new element to wrap around the Firebase storage API. Would be really useful to dinamically attach new images without need to deploy again

**Comment by cdata** [comment]
On the surface, this seems like a good place for potential improvements.

> Would be really useful to dinamically attach new images without need to deploy again

Could you please elaborate on the design you are imagining?

**Comment by stemuk** [comment]
@cdata I would probably put it like this:

`
<firebase-app auth-domain="polymerfire.firebaseapp.com"
  database-url="https://polymerfire.firebaseio.com/"
  api-key="your-api-key">
</firebase-app>
<firebase-storage-upload storage-Bucket="your-bucket-url">
</firebase-storage-upload>
`

Something similar is needed for firebase-storage-download and delete, as seen in https://firebase.google.com/docs/storage.

**Comment by hubgit** [comment]
The design of the [`vaadin-upload`](https://vaadin.com/elements/-/element/vaadin-upload) component is nice - something like that for uploading to Firebase Storage would be great.

**Comment by locomuco** [comment]
To be more specific about this the firebase app-element should be extended to accept a storage base location:

```
  <firebase-app
      name="storage-app"
      api-key="xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx"
      auth-domain="test-x.firebaseio.com"
      database-url="https://test-x.firebaseio.com/"
      storage-url="gs://test-x.appspot.com">
  </firebase-app>
```

the base storage API could look like this. 

```
  <firebase-storage
        app-name="storage-app"
        path="/pictures"
        data="image.jpg">
  </firebase-storage>
```

**Comment by stemuk** [comment]
Is there still development around this feature addition? Would love to hear some updates whether the support for firebase storage is still being designed or has been canceled. 
At least from me this addition would be highly appreciated!

**Comment by locomuco** [comment]
@stemuk please have a look in the PR #30 mentioned above, that provides a basic example for file uploads, currently I'm waiting for feedback

**Comment by stemuk** [comment]
Awesome, thanks! Hope it will be merged soon :D

**Comment by polinom** [comment]
Guys... this is so needed, please get it in asap!

**Comment by FluorescentHallucinogen** [comment]
See #226.

**Comment by webtesting2018** [comment]
Provided bucket:<App name>appspot.com does not match bucket specified in FIRApp configuration:  <app path>'

**Comment by FluorescentHallucinogen** [comment]
This issue can be closed, because pull request #226 has been merged.
